### PR TITLE
[Stripe Card] Validate unlisted designs are only issued by admins

### PR DESCRIPTION
**Pull request by garyhtou**
## Summary of the problem

Non-admins were able to issue unlisted designs. These are non-common designs that don't belong to a specific organization.

Thanks @24c02 for reporting.

## Describe your changes

Validate that only admins can issue cards with unlisted personalization designs

### CODE REVIEW

File: app/models/stripe_card.rb [hunk 1]
```diff
@@ -440,9 +440,22 @@ def only_physical_cards_can_have_personalization_design
...
+    # use the chosen design.
+
     if personalization_design&.event.present? && personalization_design.event != event
       errors.add(:personalization_design, "must be of the same event")
     end
+
+    if personalization_design.unlisted? && !user.admin?
+      # Only admins can issue unlisted designs
+      errors.add(:personalization_design, "is invalid")
+    end
```
Comment (sampoder) [inline app/models/stripe_card.rb]: It'd be nice to issue unlisted designs for others via the console, that's why this was never implemented AFAIK

**Comment by 24c02** [comment]
idk how much of an actual issue this is? like it'd be nice to fix but you have to actually guess the perso design ID
(inb4 somebody orders 8000 stripe cards ig)

File: app/models/stripe_card.rb (hunk 1 above)
Comment (garyhtou) [reply app/models/stripe_card.rb]: We could move this check into the controller to continue allowing console. I also don't think it's a big issue if people issue unlisted designs lol — so maybe we just close this pr 🤷‍♂️

Comment (sampoder) [reply app/models/stripe_card.rb]: Separate to this... I'd love an option for admins to be able to issue cards for others from the UI

**Comment by garyhtou** [comment]
@sampoder @YodaLightsabr I'll leave the final call on this up to yall. You guys have the best understanding of the unlisted feature since yall built it!

**Comment by sampoder** [comment]
I think we close this

**Comment by garyhtou** [comment]
sounds good to me!

**Pull request closed without merging**